aggregate/billing_functions/seqr.py

```diff
@@ -0,0 +1,593 @@
+                )
+            )
+
+    extra_job_resources = defaultdict(int)
+    for job in jobs_with_no_dataset:
+        job_id = job['job_id']
+        for batch_resource, usage in job['resources'].items():
+            if batch_resource.startswith('service-fee'):
+                continue
+
+            extra_job_resources[batch_resource] += usage
+
+        for batch_resource, usage in extra_job_resources.items():
+
+            batch_id = batch['id']
+            hail_ui_url = utils.HAIL_UI_URL.replace('{batch_id}', str(batch_id))
+
+            labels = {
+                'batch_name': batch_name,
+                'batch_id': str(batch_id),
+            }
+            labels['batch_resource'] = batch_resource
+            labels['url'] = hail_ui_url
+
+            gross_cost = currency_conversion_rate * utils.get_usd_cost_for_resource(
+                batch_resource, usage
+            )
+
+            for dataset, fraction in prop_map.items():
+                # Distribute the remaining cost across all datasets proportionately
+                cost = gross_cost * fraction
+
+                entries.append(
+                    utils.get_entry(
+                        key='-'.join(
+                            [
+                                SERVICE_ID,
+                                'distributed',
+                                dataset,
+                                batch_resource,
+                                'batch',
+                                str(batch_id),
+                                'job',
+                                str(job_id),
+                            ]
+                        ),
+                        topic=dataset,
+                        service_id=SERVICE_ID,
+                        description='Seqr compute (distributed)',
+                        cost=cost,
+                        currency_conversion_rate=currency_conversion_rate,
+                        usage=round(usage * fraction),
+                        batch_resource=batch_resource,
+                        start_time=start_time,
+                        end_time=end_time,
+                        labels={
+                            **labels,
+                            'dataset': dataset,
+                            'fraction': round(100 * fraction) / 100,
+                        },
+                    )
+                )
+
+    entries.extend(utils.get_hail_credits(entries))
+
+    return entries
+
+
+def migrate_entries_from_bq(
+    start, end, prop_map: ProportionateMapType, dry_run=False
+) -> int:
+    """
+    Migrate entries from BQ to GCP, using the given proportionate maps
+    """
+    # pylint: disable=too-many-branches
+    _query = f"""
+        SELECT * FROM `{GCP_BILLING_BQ_TABLE}`
+        WHERE export_time >= @start
+            AND export_time <= @end
+            AND project.id = @project
+        ORDER BY usage_start_time
+    """
+
+    job_config = bq.QueryJobConfig(
```

Comment:
TODO: this function should use the date_interval generator that the other BQ function uses